src/smc-webapp/project/new/project-new-form.tsx

```diff
@@ -0,0 +1,433 @@
+      }
+    };
+
+    submit_via_enter = (e): void => {
+      e.preventDefault();
+      this.submit();
+    };
+
+    private render_close_button(): JSX.Element | undefined {
+      const { on_close } = this.props;
+      if (!on_close) {
+        return;
+      }
+      return (
+        <Button onClick={on_close} className={"pull-right"}>
+          Close
+        </Button>
+      );
+    }
+
+    private render_error(): JSX.Element {
+      let message;
+      const error = this.props.file_creation_error;
+      if (error === "not running") {
+        message = "The project is not running. Please try again in a moment";
+      } else {
+        message = error;
+      }
+      return (
+        <ErrorDisplay
+          error={message}
+          onClose={(): void => {
+            this.props.actions.setState({ file_creation_error: "" });
+          }}
+        />
+      );
+    }
+
+    private blocked(): string {
+      if (this.props.project_map == undefined) {
+        return "";
+      }
+      if (this.props.get_total_project_quotas(this.props.project_id)?.network) {
+        return "";
+      } else {
+        return " (access blocked -- see project settings)";
+      }
+    }
+
+    private create_folder = (): void => {
+      this.props.actions.create_folder({
+        name: this.state.filename,
+        current_path: this.props.current_path,
+        switch_over: true
+      });
+      this.props.on_create_folder?.();
+    };
+
+    private render_no_extension_alert(): JSX.Element {
+      return (
+        <Alert
+          bsStyle="warning"
+          style={{ marginTop: "10px", fontWeight: "bold" }}
+        >
+          <p>
+            Warning: Are you sure you want to create a file with no extension?
+            This will use a plain text editor. If you do not want this, click a
+            button below to create the corresponding type of file.
+          </p>
+          <ButtonToolbar style={{ marginTop: "10px" }}>
+            <Button
+              onClick={(): void => {
+                this.create_file();
+              }}
+              bsStyle="success"
+            >
+              Yes, please create this file with no extension
+            </Button>
+            <Button
+              onClick={(): void => {
+                this.setState({ extension_warning: false });
+              }}
+              bsStyle="default"
+            >
+              Cancel
+            </Button>
+          </ButtonToolbar>
+        </Alert>
+      );
+    }
+
+    private render_close_row(): JSX.Element | undefined {
+      if (!this.props.on_close) {
+        return;
+      }
+      return (
+        <Row>
+          <Col sm={9}>
+            <div style={{ color: "#666" }}>
+              <em>
+                Read about{" "}
+                <a
+                  href="https://doc.cocalc.com/howto/upload.html"
+                  target="_blank"
+                  rel="noopener noreferrer"
```

Comment:
Added this line due to eslint complaining about a [security risk](https://developers.google.com/web/tools/lighthouse/audits/noopener) even though it's our own domain...

Reply:
I guess technically our doc domains might have looser code/dep review and someone could more easily slip in malicious code?